wgpu-hal/src/dx12/command.rs

```diff
@@ -1198,11 +1198,17 @@ impl crate::CommandEncoder for super::CommandEncoder {
     }
 
     unsafe fn dispatch_indirect(&mut self, buffer: &super::Buffer, offset: wgt::BufferAddress) {
-        self.prepare_dispatch([0; 3]);
-        //TODO: update special constants indirectly
+        self.update_root_elements();
+        let cmd_signature = if let Some(cmd_signatures) =
+            self.pass.layout.special_constants_cmd_signatures.as_ref()
+        {
+            &cmd_signatures.dispatch
+        } else {
+            &self.shared.cmd_signatures.dispatch
+        };
```

Comment:
style: I think it's (marginally) clearer to read the `cmd_signature` extraction this way:

```suggestion
        let cmd_signature = &self
            .pass
            .layout
            .special_constants_cmd_signatures
            .as_ref()
            .unwrap_or_else(|| &self.shared.cmd_signatures)
            .dispatch;
```

I can push this up as a `fixup!`, if you want.

Reply:
This does look better 👍

Reply:
Pushed as 8e60dcb9.